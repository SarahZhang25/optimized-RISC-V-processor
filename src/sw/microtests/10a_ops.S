// This is test 1 of 2 that tests that your processor handles all the OP
// instructions correctly.
.text
start:
    li x1, 17
    li x2, 3
    add  x3 , x1, x2
    sub  x4 , x1, x2
    slt  x5 , x1, x2
    sltu x6 , x1, x2
    slt  x7 , x2, x1
    sltu x8 , x2, x1
    sll  x9 , x1, x2
    srl  x10, x1, x2
    sra  x11, x1, x2
    and  x12, x1, x2
    or   x13, x1, x2
    xor  x14, x1, x2

    li x16, -17
    li x17, -29
    add  x18, x16, x17
    sub  x19, x16, x17
    slt  x20, x16, x17
    sltu x21, x16, x17
    slt  x22, x17, x16
    sltu x23, x17, x16
    sll  x24, x16, x17
    srl  x25, x16, x17
    sra  x26, x16, x17
    and  x27, x16, x17
    or   x28, x16, x17
    xor  x29, x16, x17
    unimp
