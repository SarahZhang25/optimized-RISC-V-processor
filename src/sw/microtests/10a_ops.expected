Dumping the state of the processor
pc = 0x00000070
x0 = 0x00000000
x1 = 0x00000011
x2 = 0x00000003
x3 = 0x00000014
x4 = 0x0000000e
x5 = 0x00000000
x6 = 0x00000000
x7 = 0x00000001
x8 = 0x00000001
x9 = 0x00000088
x10 = 0x00000002
x11 = 0x00000002
x12 = 0x00000001
x13 = 0x00000013
x14 = 0x00000012
x15 = 0x00000000
x16 = 0xffffffef
x17 = 0xffffffe3
x18 = 0xffffffd2
x19 = 0x0000000c
x20 = 0x00000000
x21 = 0x00000000
x22 = 0x00000001
x23 = 0x00000001
x24 = 0xffffff78
x25 = 0x1ffffffd
x26 = 0xfffffffd
x27 = 0xffffffe3
x28 = 0xffffffef
x29 = 0x0000000c
x30 = 0x00000000
x31 = 0x00000000
Quitting simulation.
